osu.Game.Rulesets.Osu/Difficulty/Skills/Speed.cs

```diff
@@ -20,20 +22,119 @@ public class Speed : Skill
         private const double pi_over_4 = Math.PI / 4;
         private const double pi_over_2 = Math.PI / 2;
 
+        /// <summary>
+        /// streamaimconst is derived from the last stream of Megalovania [Excors] 9* map (Unit:pixel/ms)
+        /// </summary>
+        private const double streamaimconst = 2.42;
```

Comment:
Okay, so here's probably one of the core issues of this whole rework. You choose an arbitrary constant based on *one* beatmap. Why this one? What does it have to do with difficulty in particular? Why are other maps not taken into account?

What you seem to want to do is to monkey-patch the existing system to get pp to be more in line with what you consider best. I think a more holistic approach to the *entire* difficulty system might be better.

Reply:
Okay, that is the ceiling value for the speed which seperates spaced streams from jumps. Excors last stream section has the fastest spaced stream section near the end so I base this value on that. If you've seen faster tell me so that I can fix it.

Reply:
I'll only buff fast velocity with positive acceleration streams.

Reply:
> I'll only buff fast velocity with positive acceleration streams.
That is the reason for code complexity as well.

Reply:
> Excors last stream section has the fastest spaced stream section near the end so I based this value on that.

And this is the whole issue - you're not supposed to rework the whole system based on what happens in *one* specific map. What happens if someone ranks a map with much more spaced streams? Do we have to change this value every single time the 'fastest ranked stream' changes?

A proper PP rework needs to be based on explainable formulas, with their potential effects on PP changes carefully studied for all kinds of maps.

Seeing the discussion which took place in the #difficulty-osu Discord channel, these changes were developed by taking a few hard maps from top ranks of high-level players and throwing around random solutions that happened to work on those specific maps. 

There were situations where values had to be adjusted to fit expectations on *one* specific map (or even play) and I think that shows best why it's a fundamentally flawed approach - you have to take *all* kinds of maps and plays into account, both very easy and very hard, plus everything inbetween, having a reason for every single change and making sure that it will work universally, not just for a very specific set of beatmaps.

Reply:
Pay for my cellular data, tell me how to use osu-tools to aggregate data from data.ppy.sh and I'll do that for you.

Reply:
$2.31/4GB of data

Reply:
And no, I'm not being salty or sarcastic here, I've literally paid about $50 for data due to development reasons.

Reply:
So let me get this straight, you made a proposal (unlike the already increasing amount of ideas) which uses arbitrary values, that would be alright if not for some evaluation and a lot of evidence. You used one [specific beatmap](https://osu.ppy.sh/beatmapsets/375201#osu/821785) just for spaced streams, which is loved and, yes there are other beatmaps that could have larger spaced streams. 
Also, your comeback to WildTree is a bit absurd, you requested the contributors to pay you $183 for a download of all of the osu tar files? That does sound like you were expecting everyone to help you out? They are just here to check over and see if it's good enough quality to be in the game. Well about that..

After checking the issues that are brought up, I'm guessing you're just a beginner at developing with osu! and tried your way to merge it into osu!lazer? (You were supposed to pull request on osu-performance), The whole system developed right now is supposed to be complicated, it has to have a flexible threshold just in case you have out-there maps that you didn't account for. 
You could have used a distance spacing threshold that depended on the bpm or density of the entire song/section.  Or a more complex method that I wouldn't have thought of. But right now, the tree is nowhere NEAR merge-able due to the incoherent equations/code (otherwise known as spaghetti code which is not a good thing), inadequate evidence to show your proposal actually working, multiple rookie mistakes along the way and some others.